app/controllers/nodes_controller.rb

```diff
@@ -42,12 +42,11 @@ def update
   # assigned role, assign a random role to it, and then call the salt
   # orchestration.
   def bootstrap
-    if Minion.where(role: nil).count > 1
-      # choose first minion to be the master
-      Minion.assign_roles!(roles: { Minion.first.hostname => ["master"] })
+    if Minion.all.collect(&:role).include?("master")
```

Comment:
This could be more rails-like: `Minion.exists?(role: :master)`, it's faster too.